### ISSUE TITLE
"define" or "begin" clashing with a keyword

### ISSUE THREAD

**Issue by rebcabin** [Calysto/calysto_scheme#39]
Consider the following, which came up while writing a metacircular evaluator in calysto:

```scheme
(case 'begin
  ((begin) 'foo-begin)
  ((define) 'foo-define)
  (else 'foo-else))
```

In Gambit Scheme, this produces

```bash
Gambit v4.9.4-132-gb9304a3b

> (case 'begin
  ((begin) 'foo-begin)
  ((define) 'foo-define)
  (else 'foo-else))
foo-begin
```

In calysto Scheme, this produces a parse error (I've run "test_all.ss" in the same REPL session to show that everything else seems OK):

```bash
└─(15:25:53 on master ✹ ✚ ✭)──> python scheme.py                                                                                                      2 ↵ ──(Tue,Dec27)─┘
Calysto Scheme, version 1.4.7
----------------------------
Use (exit) to exit
==> (case 'begin
  ((begin) 'foo-begin)
  ((define) 'foo-define)
  (else 'foo-else))... ... ... 
... 

Traceback (most recent call last):
ParseError: bad concrete syntax: (begin)
==> (load "test_all.ss")
Testing group 'defines'...
Testing group 'range'...
Testing group 'sort'...
Testing group 'main'...
Testing group 'mu-lambda'...
Testing group 'define'...
Testing group 'call/cc'...
Testing group 'try'...
Testing group 'loop'...
Testing group 'macros'...
Testing group 'datatype'...
Testing group 'named-parameters'...
=================
Testing completed!
  Time : 0.37 seconds
  Total tests defined: 247 
  Total tests tested : 247 
                Right: 247 
                Wrong: 0 
==> 
```

**Comment by rebcabin** [comment]
Note that the following works in Calysto, so my guess is that Calysto is misinterpreting the symbols `begin` and `define` in the context of `case`:

```scheme
(case 'beginx
  ((beginx) 'foo-beginx)
  ((definex) 'foo-definex)
  (else 'foo-else))
```

```bash
└─(15:32:27 on master ✹ ✚ ✭)──> python scheme.py                                                                                                          ──(Tue,Dec27)─┘
Calysto Scheme, version 1.4.7
----------------------------
Use (exit) to exit
==> (case 'beginx
  ((beginx) 'foo-beginx)
  ((definex) 'foo-definex)
  (else 'foo-else))
... ... ... foo-beginx
==>
```

**Comment by dsblank** [comment]
Fixed in Calysto Scheme 1.4.8. The issue was in how we applied an internal thunk. Now we specifically use `apply`. Thanks for reporting!
